source/LibRender2/Textures/TextureManager.cs

```diff
@@ -198,29 +203,69 @@ public Texture RegisterTexture(Bitmap bitmap, Bitmap alpha)
 		/// <returns>Whether loading the texture was successful.</returns>
 		public bool LoadTexture(Texture handle, OpenGlTextureWrapMode wrap, int currentTicks, InterpolationMode Interpolation, int AnisotropicFilteringLevel)
 		{
+
+			Texture texture = null;
 			//Don't try to load a texture to a null handle, this is a seriously bad idea....
 			if (handle == null || handle.OpenGlTextures == null)
 			{
 				return false;
 			}
-
+			
+			if (handle.MultipleFrames)
+			{
+				if (!(animatedTextures.ContainsKey(handle.Origin)))
+				{
+					if (!handle.Origin.GetTexture(out texture))
+					{
+						//Loading animated texture barfed
+						return false;
+					}
+					animatedTextures.Add(handle.Origin, texture);
+				}
+				else
+				{
+					texture = animatedTextures[handle.Origin];
+				}
+				double elapsedTime = CPreciseTimer.GetElapsedTime(handle.LastAccess, currentTicks);
+				int elapsedFrames = (int)(elapsedTime / handle.FrameInterval);
+				if (elapsedFrames > 0)
+				{
+					handle.CurrentFrame += elapsedFrames;
+					handle.LastAccess = currentTicks;
+					for (int i = 0; i < 4; i++)
+					{
+						GL.DeleteTexture(handle.OpenGlTextures[i].Name);
+						handle.OpenGlTextures[i].Valid = false;
```

Comment:
Optimise- Can probably use a jagged array to generate the GL textures for all frames on demand and cache